source/localizable/components/triggering-changes-with-actions.md

```diff
@@ -245,76 +245,90 @@ In this case, the code in `button-with-confirmation` does not change.
+We want to arrange for the action to be invoked from within `button-with-confirmation` with both arguments.
+A feature of action arguments that applies to this situation is that they _curry_.
+In general (not just in the context of Ember), this refers to the idea that a function with multiple arguments can be expressed as a sequence of functions each taking one argument and returning the next function in the sequence that takes the next argument,
+and so on.
+Evaluating the last function produced in this way is equivalent to evaluating the original function with all of its arguments.
 
-```app/templates/components/send-message.hbs
-{{#button-with-confirmation
-    text="Click to send your message."
-    onConfirm=(action "sendMessage" "info")
-    as |confirmValue|}}
-  {{input value=confirmValue}}
-{{/button-with-confirmation}}
-```
+For our purposes, currying means that when we pass the action to a component, if we provide the action helper with a subset of arguments (e.g. by providing the value of `messageType` when we pass `sendMessage` to `button-with-confirmation`), the result of the helper (in our case `onConfirm`) is a function that can be invoked within the component javascript file with just the remaining arguments,
+in our case `messageText`.
 
-Now when the `submitConfirm` action is invoked, we call it with the value provided by our yielded input.
+Remember, however,
+that our `button-with-confirmation` component does not know or care that it is being used here in a messaging application.
+So to enable it to call the action represented by `onConfirm` with the required remaining argument,
+we will do three things:
 
-```app/components/button-with-confirmation.js
-import Ember from 'ember';
+_First_, in the component we will use a property `confirmValue` to contain a parameter to be passed to `onConfirm`:
 
+```app/components/button-with-confirmation.js
 export default Ember.Component.extend({
   actions: {
-    launchConfirmDialog() {
-      this.set("confirmShown", true);
-    },
-
+    //...
     submitConfirm() {
-      //call onConfirm with the value of the input field as an argument
+      //call onConfirm with a second argument
       const promise = this.get('onConfirm')(this.get('confirmValue'));
       promise.then(() => {
         this.set('confirmShown', false);
       });
     },
-
-    cancelConfirm() {
-      this.set('confirmShown', false);
-    }
+    //...
   }
 });
 ```
 
-This action will call our bound `sendMessage` function with both the message type we provided earlier, and the template
-and the message value provided in the component JavaScript.
+When we invoke `onConfirm` with this as an explicit argument,
+because of currying that argument will be passed to `sendMessage` as its _second_ argument.
+Although in our particular example, `confirmValue` will be used to pass the message text,
+using this ensures that the implementation of the component remains independent of the specific purpose it is being used for.
```

Comment:
Here again, mentioning currying may actually reduce clarity here.  Maybe something like this?

> Because the first argument needed in our `sendMessage` action has already been filled in in the `send-message` template, when we call `onConfirm` our `confirmValue` property will be be used as the second argument.  This makes it possible to re-use our `button-with-confirmation` component in different places where the action it calls may have very different method signatures.

Even my suggestion here may need further wordsmithing, but I think it conveys the same concepts in simpler language.